spec/requests/swagger_spec.rb

```diff
@@ -0,0 +1,454 @@
+      it 'returns a HTTP 403 response' do
+        login
+        api_key = FakeGitHub::INVALID_API_KEY
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions',
+                                    403,
+                                    "_headers" => headers(api_key))
+      end
+    end
+
+    context 'when the API key is missing' do
+      it 'returns a HTTP 403 response' do
+        login
+        api_key = nil
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions',
+                                    403,
+                                    "_headers" => headers(api_key))
+      end
+    end
+
+    context 'when the user is not an admin' do
+      it 'returns a HTTP 403 response' do
+        login
+        api_key = FakeGitHub::VALID_API_KEY
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions',
+                                    403,
+                                    "_headers" => headers(api_key))
+      end
+    end
+
+    context 'when the user is an admin' do
+      let(:user) { FactoryGirl.create(:admin_user) }
+
+      it 'returns a 200 HTTP response' do
+        login(user)
+        api_key = FakeGitHub::VALID_API_KEY
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions',
+                                    200,
+                                    "_headers" => headers(api_key))
+      end
+
+      it 'returns valid auctions' do
+        login(user)
+        create(:auction, :with_bidders)
+        api_key = FakeGitHub::VALID_API_KEY
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions',
+                                    200,
+                                    "_headers" => headers(api_key))
+      end
+    end
+  end
+
+  context 'GET /admin/auctions/{id}' do
+    context 'when the API key is invalid' do
+      it 'returns a 403 HTTP response' do
+        login
+        api_key = FakeGitHub::INVALID_API_KEY
+        auction = create(:auction, :with_bidders)
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions/{id}',
+                                    403,
+                                    "id" => auction.id, "_headers" => headers(api_key))
+      end
+    end
+
+    context 'when the API key is missing' do
+      it 'returns a 403 HTTP response' do
+        login
+        api_key = nil
+        auction = create(:auction, :with_bidders)
+
+        expect(subject).to validate(:get,
+                                    '/admin/auctions/{id}',
+                                    403,
+                                    "id" => auction.id, "_headers" => headers(api_key))
+      end
+    end
+
+    context 'when the API key is valid' do
+      it 'returns a 200 HTTP response' do
+        user = FactoryGirl.create(:admin_user)
```

Comment:
seems like we're using `FactoryGirl.create` and `create` in here - fine with either, but can we stick with one or the other per spec file?